test/perf/tablet_load_balancing.cc

```diff
@@ -301,6 +305,7 @@ future<results> test_load_balancing_with_many_tables(params p, bool tablet_aware
 
                 res.tables[table_index++] = {
                     .shard_overcommit = shard_overcommit,
+                    .best_shard_overcommit = best_shard_overcommit,
```

Comment:
We could improve the "best possible shard overcommit" by making the tablet size goal proportional to the space actively used, not the total space (and right now we have a fixed 5GB goal, so it's not even proportional to total space.

So, if a shard has 10GB space used, then tablet size goal is 100MB, and we always have 100 tablets/shard.  If there's one table and it starts to grow, then the goal will grow with it and there will be no split. If there are multiple tables and they grow disproportionately, there will be split/merge as required to equalize the tablet size.

This solves the grow-from-nothing problem. It makes count(*) on an empty cluster slower, but I think it's a good tradeoff.

/cc @raphaelsc

Reply:
But this means that smaller tables will have fewer tablets, so less granularity, which leads to higher shard overcommit. In the extreme case, the whole table may be put in one tablet.

Reply:
We have the same problem with 5GB tablets. If a table uses less than 5GB, it will be on one shard.

If anything it makes the problem smaller, since clusters with a small amount of active storage will have smaller tablets.

We still need to solve the case of a very small table that lives alongside a very large table, but it's possible to come up with heuristics for that.

(note that many-small-tables will have to be satisfied with not using the entire cluster)